src/grisp_onewire.erl

```diff
@@ -80,23 +121,56 @@ write_config(Conf) when is_integer(Conf) ->
         Any -> error({read_back_config, Any, Val})
     end.
 
+%% @doc Reset device and activate active pullup (APU).
+%%
+%% This function can only be used inside of a {@link transaction/1} call.
+%% @see reset/0.
+%% @see write_config/1.
+%% @end
+-spec detect() -> 'ok'.
 detect() ->
     assert_transaction(),
     reset(),
     write_config([apu]).
 
+%% @doc Reset the bus and check the register for devices.
+%%
+%% Command code: b4h.
+%%
+%% This function can only be used inside of a {@link transaction/1} call.
+%%
+%% === Return Value to Register Bit Mapping ===
+%% <table border="1" summary="Return Value to Register Mapping">
+%%  <tr><th>Atom</th><th>Bit 1 PPD</th><th>Bit 2 SD</th></tr>
+%%  <tr><td>`nothing_present'</td><td>0</td><td>0</td></tr>
+%%  <tr><td>`presence_detected'</td><td>1</td><td>0</td></tr>
+%%  <tr><td>`short_detected'</td><td>0</td><td>1</td></tr>
```

Comment:
Rather than the bit mapping maybe explaining the return values could be added.  Same for write config.

Here its easier that for the config:  

nothing_detected -> no device on bus detected
presence_dectected -> some devices on the bus
short_detected -> there is a short circuit between data and ground on the bus detected (thats a failure)